src/families/ripple/js-synchronization.ts

```diff
@@ -0,0 +1,182 @@
+import BigNumber from "bignumber.js";
+import {
+  getAccountInfo,
+  getServerInfo,
+  getTransactions,
+} from "../../api/Ripple";
+import {
+  GetAccountShape,
+  makeScanAccounts,
+  makeSync,
+  mergeOps,
+} from "../../bridge/jsHelpers";
+import { encodeOperationId } from "../../operation";
+import { Account, Operation } from "../../types";
+import { encodeAccountId } from "../../account";
+import { NEW_ACCOUNT_ERROR_MESSAGE } from "./bridge/js";
+
+interface Currency {
+  currency: string;
+  amount: string;
+}
+
+interface TxXRPL {
+  meta: {
+    TransactionResult: string;
+    delivered_amount: Currency | string;
+  };
+  tx: {
+    TransactionType: string;
+    Fee: string;
+    Account: string;
+    Destination: string;
+    DestinationTag?: number;
+    Amount: string;
+    Sequence: number;
+    date: number;
+    inLedger: number;
+    hash: string;
+  };
+}
+
+const txToOperation =
+  (accountId: string, address: string) =>
+  ({
+    meta: { delivered_amount },
+    tx: {
+      DestinationTag,
+      Fee,
+      hash,
+      inLedger,
+      date,
+      Account,
+      Destination,
+      Sequence,
+    },
+  }: TxXRPL): Operation | null | undefined => {
+    const type = Account === address ? "OUT" : "IN";
+    let value =
+      delivered_amount && typeof delivered_amount === "string"
+        ? new BigNumber(delivered_amount)
+        : new BigNumber(0);
+    const feeValue = new BigNumber(Fee);
+
+    if (type === "OUT") {
+      if (!Number.isNaN(feeValue)) {
+        value = value.plus(feeValue);
+      }
+    }
+
+    // https://xrpl.org/basic-data-types.html#specifying-time
+    const toEpochDate = (946684800 + date) * 1000;
+
+    const op: Operation = {
+      id: encodeOperationId(accountId, hash, type),
+      hash: hash,
+      accountId: accountId,
+      type,
+      value,
+      fee: feeValue,
+      blockHash: null,
+      blockHeight: inLedger,
+      senders: [Account],
+      recipients: [Destination],
+      date: new Date(toEpochDate),
+      transactionSequenceNumber: Sequence,
+      extra: {},
+    };
+
+    if (DestinationTag) {
+      op.extra.tag = DestinationTag;
+    }
+
+    return op;
+  };
+
+const filterOperations: any = (
+  transactions: TxXRPL[],
+  accountId: string,
+  address: string
+) => {
+  return transactions
+    .filter(
+      (tx: TxXRPL) =>
+        tx.tx.TransactionType === "Payment" &&
+        typeof tx.meta.delivered_amount === "string"
+    )
+    .map(txToOperation(accountId, address))
+    .filter(Boolean);
+};
+
+const getAccountShape: GetAccountShape = async (
+  info
+): Promise<Partial<Account>> => {
+  const { address, initialAccount, currency, derivationMode } = info;
+  const accountId = encodeAccountId({
+    type: "js",
+    version: "2",
+    currencyId: currency.id,
+    xpubOrAddress: address,
+    derivationMode,
+  });
+  const accountInfo = await getAccountInfo(address);
+
+  if (!accountInfo || accountInfo.error === NEW_ACCOUNT_ERROR_MESSAGE) {
```

Comment:
When an account is not created or exist, the api will return this NEW_ACCOUNT_ERROR_MESSAGE and we will not get any info so yeah, in case of scan I guess we should return a shallow account